compiler/rustc_typeck/src/astconv/generics.rs

```diff
@@ -43,23 +43,64 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             }
         }
 
+        let add_braces_suggestion = |arg: &GenericArg<'_>, err: &mut DiagnosticBuilder<'_>| {
+            let suggestions = vec![
+                (arg.span().shrink_to_lo(), String::from("{ ")),
+                (arg.span().shrink_to_hi(), String::from(" }")),
+            ];
+            err.multipart_suggestion(
+                "if this generic argument was intended as a const parameter, \
+                   surround it with braces:",
+                suggestions,
+                Applicability::MaybeIncorrect,
+            );
+        };
+
         // Specific suggestion set for diagnostics
         match (arg, &param.kind) {
             (
-                GenericArg::Type(hir::Ty { kind: hir::TyKind::Path { .. }, .. }),
-                GenericParamDefKind::Const { .. },
+                GenericArg::Type(hir::Ty {
+                    kind: hir::TyKind::Path(rustc_hir::QPath::Resolved(_, path)),
+                    ..
+                }),
+                GenericParamDefKind::Const,
             ) => {
-                let suggestions = vec![
-                    (arg.span().shrink_to_lo(), String::from("{ ")),
-                    (arg.span().shrink_to_hi(), String::from(" }")),
-                ];
-                err.multipart_suggestion(
-                    "if this generic argument was intended as a const parameter, \
-                try surrounding it with braces:",
-                    suggestions,
-                    Applicability::MaybeIncorrect,
-                );
+                use rustc_hir::def::{DefKind, Res};
+                match path.res {
+                    Res::Err => {
+                        add_braces_suggestion(arg, &mut err);
+                        err.set_primary_message(
+                            "unresolved item provided when a constant was expected",
+                        )
+                        .emit();
+                        return;
```

Comment:
Why is the early `return` needed? Not saying it's wrong, just want to understand what it.

Reply:
Ah I don't think it's needed, but it just guarantees we don't emit the swapped order suggestion below because it won't be relevant for an error type

Reply:
Could you try coming up with a test case that triggers both of those case?

Reply:
Hm I'm not actually sure I can run into this case currently, because types must declared as parameters before consts specifically within min-const-generics, and this branch only runs when there is a type where a const was expected.

Should I just remove the return then?